Drop-In/src/main/java/com/braintreepayments/api/SupportedPaymentMethodsFragment.java

```diff
@@ -111,19 +106,22 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
             }
         });
 
+        dropInViewModel.getUserCanceledError().observe(getViewLifecycleOwner(), exception -> {
+            if(exception instanceof UserCanceledException){
+                setViewState(ViewState.SHOW_PAYMENT_METHODS);
+            }
+        });
+
         vaultManagerButton.setOnClickListener(v -> sendDropInEvent(new DropInEvent(DropInEventType.SHOW_VAULT_MANAGER)));
 
+        cancelButton.setOnClickListener(v -> dropInViewModel.setBottomSheetState(BottomSheetState.HIDE_REQUESTED));
         sendAnalyticsEvent("appeared");
         return view;
     }
 
     @Override
     public void onResume() {
         super.onResume();
-
-        if (viewState == ViewState.LOADING && hasSupportedPaymentMethods()) {
```

Comment:
This didn't seem to be necessary - the loading/cancellation flows all seem to be working correctly without this logic. Do we need it?